src/visualization/types/Map/view.tsx

```diff
@@ -21,51 +27,97 @@ const GeoPlot: FC<Props> = ({result, properties}) => {
     detectCoordinateFields,
     mapStyle,
   } = properties
-  const tileServerConfiguration = {
-    tileServerUrl: '',
-    bingKey: '',
-  }
 
   const {lat, lon} = properties.center
 
-  let calculatedGeoCoordinates = {
+  const [errorCode, setErrorCode] = useState<string>('')
+  const [mapToken, setMapToken] = useState<string>('')
+  const [geoCoordinates, setGeoCoordinates] = useState<Coordinates>({
     lat,
     lon,
-  }
-  try {
-    calculatedGeoCoordinates = getGeoCoordinates(result.table, 0)
-  } catch (err) {
-    const error =
-      'Map type is not supported with the data provided: Missing latitude/longitude values'
+  })
+
+  useEffect(() => {
+    const getToken = async () => {
+      try {
+        const {token} = await getMapToken()
+        setMapToken(token)
+      } catch (err) {
+        setErrorCode('MapServiceUnavailable')
+      }
+    }
+    const getCoordinates = () => {
```

Comment:
I think you can pull this function out of the `useEffect` call and leave it where it was since it's not a side effect

Reply:
I pulled this out like before